### ISSUE TITLE
block is not consistent with `df`

### ISSUE THREAD

**Issue by chiptus** [jaypipes/ghw#57]
For some reason, while I have only 256GB, running the block example code shows I have 2TB.

running `df -h`:
![screenshot from 2018-09-18 13-10-48](https://user-images.githubusercontent.com/1381655/45680728-4e433e00-bb44-11e8-8e7c-584f3bcd072e.png)

running the following:
```
ackage main

import (
	"fmt"

	"github.com/jaypipes/ghw"
)

func main() {
	block, err := ghw.Block()
	if err != nil {
		fmt.Printf("Error getting block storage info: %v", err)
	}

	fmt.Printf("%v\n", block)

	for _, disk := range block.Disks {
		fmt.Printf(" %v\n", disk)
		for _, part := range disk.Partitions {
			fmt.Printf("  %v\n", part)
		}
	}
}
```

![screenshot from 2018-09-18 13-12-20](https://user-images.githubusercontent.com/1381655/45680778-8185cd00-bb44-11e8-8a3f-90caff40d9b4.png)

**Comment by jaypipes** [comment]
Hi @chiptus, that's odd. It seems to be indicating that ghw thinks the output of your first disk's `String()` method is "204848114432". I have no idea why that would be, considering `ghw.Disk.String()`'s implementation looks like this:

```go
func (d *Disk) String() string {
	vendor := ""
	if d.Vendor != "" {
		vendor = "  " + d.Vendor
	}
	serial := ""
	if d.SerialNumber != "" {
		serial = " - SN #" + d.SerialNumber
	}
	sizeStr := "unknown"
	if d.SizeBytes > 0 {
		size := d.SizeBytes
		unit, unitStr := unitWithString(int64(size))
		size = uint64(math.Ceil(float64(size) / float64(unit)))
		sizeStr = fmt.Sprintf("%d%s", size, unitStr)
	}
	return fmt.Sprintf(
		"/dev/%s (%s) [%s]%s%s",
		d.Name,
		sizeStr,
		d.BusType,
		vendor,
		serial,
	)
}
```

In order to diagnose what's going on, could you paste the output of the following on your machine please?

```
cat /sys/block/sda/queue/physical_block_size
cat /sys/block/sda/size
cat /sys/block/sda/vendor
```

Thanks in advance!
-jay

**Comment by chiptus** [comment]
- `cat /sys/block/sda/queue/physical_block_size`: 4096
- `cat /sys/block/sda/size`: 500118192
- `cat /sys/block/sda/vendor`: `No such file or directory`

**Comment by jaypipes** [comment]
@chiptus OK, I figured it out (I think!). Turns out Linux *always* thinks the disk sector size is 512 bytes and what is stored in /sys/block/sda/size is the number of 512-byte sectors (according to Linux that is). Your *actual* physical block size is 4K, not 512 bytes, and `ghw`'s code is erroneously using the actual physical block size times the /sys/block/sda/size amount to calculate the total physical bytes of storage, instead of hard-coding 512-byte sector size.
